WIP Website v0 working with latest ocular-gatsby

chrisirhc opened the pull request.

Xiot commented:
> Gatsby issue where it's now allowing imports with import not in the top:
The issue looks like its not transpiling the gatsby code.
Also, I would think that code from gatsby shouldn't end up in the "compiled bundle" (`es` and `dist` folder)

Other than that this looks like a good path forward.

chrisirhc commented:
> > Gatsby issue where it's now allowing imports with import not in the top:
> > The issue looks like its not transpiling the gatsby code.
> > Also, I would think that code from gatsby shouldn't end up in the "compiled bundle" (`es` and `dist` folder)
> 
> Other than that this looks like a good path forward.

It's actually reading from the compiled bundle, `es` folder. It's not ending up in that folder.

ibgreen commented:
> Examples are now working. Added one working example and have MDX plugin working

@chrisirhc Very happy to see ocular being used here, and very interested in the MDX integration - we need it in other places as well.

A quick note - ocular has now been repackaged as a gatsby-theme (`gatsby-theme-ocular`) which is a considerably cleaner solution, as ocular is now just another plugin in the config, see e.g: https://github.com/visgl/loaders.gl/blob/master/website/gatsby-config.js

If react-vis moved to use the new theme, we'd be able to share ocular and gatsby config improvements more easily across projects.  It would also make react-vis feel more like a visgl project, allow visgl maintainers to (potentially) assist here.

chrisirhc commented:
> > Examples are now working. Added one working example and have MDX plugin working
> 
> @chrisirhc Very happy to see ocular being used here, and very interested in the MDX integration - we need it in other places as well.
> 
> A quick note - ocular has now been repackaged as a gatsby-theme (`gatsby-theme-ocular`) which is a considerably cleaner solution, as ocular is now just another plugin in the config, see e.g: https://github.com/visgl/loaders.gl/blob/master/website/gatsby-config.js
> 
> If react-vis moved to use the new theme, we'd be able to share ocular and gatsby config improvements more easily across projects. It would also make react-vis feel more like a visgl project, allow visgl maintainers to (potentially) assist here.

I updated the PR to leverage the ocular theme and also updated this with the new MDX approach.

See:
https://github.com/chrisirhc/react-vis/blame/website-v0/docs/scales-and-data.mdx#L12-L14

There's still quite a bit of remaining work to clean up integration with the ocular theme. I needed to copy-paste the logic from on-create-node, and create-pages (making some changes) in order to get this into a half-working state.
